gcc/d/patches/patch-versym-os-8.x

```diff
@@ -137,27 +166,14 @@ These official OS versions are not implemented:
 +#define TARGET_OS_D_BUILTINS() 			\
 +  do 						\
 +    {						\
-+      TARGET_GENERIC_LINUX_OS_D_BUILTINS();	\
++      GNU_USER_TARGET_OS_D_BUILTINS();		\
 +      ANDROID_TARGET_OS_D_BUILTINS();		\
 +    }						\
 +  while (0)
 +
  #undef CC1_SPEC
  #define CC1_SPEC \
    LINUX_OR_ANDROID_CC (GNU_USER_TARGET_CC1_SPEC, \
---- a/gcc/config/i386/mingw-w64.h
-+++ b/gcc/config/i386/mingw-w64.h
-@@ -89,3 +89,10 @@ along with GCC; see the file COPYING3.  If not see
-   %{static:-Bstatic} %{!static:-Bdynamic} \
-   %{shared|mdll: " SUB_LINK_ENTRY " --enable-auto-image-base} \
-   %(shared_libgcc_undefs)"
-+
-+#undef TARGET_OS_D_BUILTINS
-+#define TARGET_OS_D_BUILTINS()					\
-+    do {							\
-+      TARGET_GENERIC_MINGW_OS_D_BUILTINS();			\
-+      builtin_define ("GNU_MinGW64");				\
-+    } while (0)
 --- a/gcc/config/i386/mingw32.h
```

Comment:
> Well, if you need it, I can add it back in.

No need to, I don't think we need the distinction anywhere.

> By the way, I think we should also be declaring CRuntime_Microsoft for MinGW

I think so, though I never 100% understood the MinGW/MSVCRT relationship. AFAIK MinGW runtime is a combination of
* An old version of the microsoft runtime (`msvcrt.dll`). Specific toolchains can also use newer versions. Additionally future versions may use the universal crt, ucrt at some time.
* libgcc
* Additional libraries to implement C99, pthreads, ...

So we actually use a superset at `CRuntime_Microsoft` but our MSVCRT is much older than the one DMD links to.